ffc/codegeneration/definitions.py

```diff
@@ -16,35 +16,13 @@
 def num_coordinate_component_dofs(coordinate_element):
     """Get the number of dofs for a coordinate component for this degree.
 
-    This is a local hack that works for Lagrange 1-3, better
-    would be to get this passed by ffc from fiat through the ir.
-    The table data is to messy to figure out a clean design for that quickly.
     """
-    from ufl.cell import num_cell_entities
-    degree = coordinate_element.degree()
-    cell = coordinate_element.cell()
-    tdim = cell.topological_dimension()
-    cellname = cell.cellname()
-    d = 0
-    for entity_dim in range(tdim + 1):
-        # n = dofs per cell entity
-        if entity_dim == 0:
-            n = 1
-        elif entity_dim == 1:
-            n = degree - 1
-        elif entity_dim == 2:
-            if cellname == "quadrilateral":
-                n = (degree - 1) * (degree - 1)
-            else:
-                n = (degree - 2) * (degree - 1) // 2
-        elif entity_dim == 3:
-            n = (degree - 3) * (degree - 2) * (degree - 1) // 6
-        else:
-            raise RuntimeError("Entity dimension out of range")
-        # Accumulate
-        num_entities = num_cell_entities[cellname][entity_dim]
-        d += num_entities * n
-    return d
+    from ffc.fiatinterface import create_element
```

Comment:
I thought flake8 was supposed to stop you importing in the middle of a file...?

Reply:
Good question. Have moved it regardless